### DISCUSSION TITLE
Welcome to carbon-web-components Discussions!

### DISCUSSION THREAD

**Discussion by asudoh** [carbon-design-system/carbon-web-components#557] [General]
## 👋 Welcome!
  We’re using Discussions as a place for IBMers and community members to:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Solve issues.
  * Learn what the core team is planning
  * And more!

**Comment by TrentBrown** [comment]
When I follow the instructions in the top-level README file to install the component library:

`npm install -S carbon-web-components carbon-components`

I find that it leads to a peer dependency conflict:

```
✗ npm install --save carbon-web-components carbon-components
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: ft-widget@1.0.0
npm ERR! Found: carbon-components@10.34.0
npm ERR! node_modules/carbon-components
npm ERR!   carbon-components@"^10.34.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer carbon-components@"~10.32.0" from carbon-web-components@1.13.1
npm ERR! node_modules/carbon-web-components
npm ERR!   carbon-web-components@"*" from the root project
```

Looking at the change history of the project's package.json file, I see that the version of `carbon-components` library dependency is locked down (using the tilde version specifier) and is periodically bumped up. Presumably somebody is carefully testing that the `carbon-web-components` library is in fact compatible with the newer version of the `carbon-components` library. But is it truly necessary to lock down versions that tightly? If it is, people should be told to specify the versions explicitly in the `npm install` comand:

`npm install --save carbon-components@10.32.0 carbon-web-components@1.13.1`
 
That works, but specifying versions in this way is not something that I can recall seeing before. It seems like something outside of best practices.

Also: The use of the "-S" option in the README's npm command example no longer works. Would be good to change:

`npm install -S carbon-web-components carbon-components lit-html lit-element`

to:

`npm install --save carbon-web-components carbon-components lit-html lit-element`

**Reply by asudoh** [reply]
Created #634. Wrt `ERESOLVE` above, `--legacy-peer-deps` seems the quickest help. The team can decide what to do with NPM7's new behavior that auto-installs peer-dependencies (CC @jeffchew). The `carbon-components` version range is not much strict (it's basically for JavaScript `import`s seen in `carbon-web-components/src`), but from what the log above tells, `carbon-web-components` supporting the latest `carbon-components` seems the most desirable (the team needs to upgrade `carbon-components` in `devDependencies` and run visual regression tests, etc. to do so).
